src/libraries/Microsoft.PowerFx.Core/Public/Types/TypeCheckers/FormulaTypeChecker.cs

```diff
@@ -0,0 +1,101 @@
+﻿// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+using System;
+using System.Collections.Generic;
+using Microsoft.PowerFx.Core.Types;
+using Microsoft.PowerFx.Types;
+
+namespace Microsoft.PowerFx.Core.Public.Types.TypeCheckers
+{
+    internal abstract class FormulaTypeChecker
+    {
+        protected ICollection<ExpressionError> _errorList;
+        protected readonly AggregateTypeChecker _aggregateTypeChecker;
+
+        public bool Run(FormulaType sourceType, FormulaType typeToCheck)
+        {
+            if (sourceType == null || typeToCheck == null)
+            {
+                return false;
+            }
+
+            if (sourceType._type == DType.ObjNull || typeToCheck._type == DType.ObjNull)
+            {
+                return true;
+            }
+
+            // check Aggregate types
+            if (sourceType._type.IsAggregate && typeToCheck._type.IsAggregate)
```

Comment:
>sourceType._type.IsAggregate [](http://example.com/codeflow?start=16&length=28)

nit: sourceType is AggregateType sourceType

avoids the cast on 31

Reply:
updated.